pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -1830,6 +1830,59 @@ func TestNeedsRemediation(t *testing.T) {
 			expectedNextCheck:           time.Duration(1 * time.Minute), // 300-200 rounded
 			expectedError:               false,
 		},
+		{
+			testCase: "unhealthy: machine reached its maximal age",
+			target: &target{
+				Machine: mapiv1beta1.Machine{
+					TypeMeta: metav1.TypeMeta{Kind: "Machine"},
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations:       make(map[string]string),
+						Name:              "machine",
+						Namespace:         namespace,
+						Labels:            map[string]string{"foo": "bar"},
+						OwnerReferences:   []metav1.OwnerReference{{Kind: "MachineSet"}},
+						CreationTimestamp: metav1.Time{Time: time.Now().AddDate(0, 0, -1)},
+					},
+					Spec: mapiv1beta1.MachineSpec{},
+				},
+				Node: &corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "node",
+						Namespace: metav1.NamespaceNone,
+						Annotations: map[string]string{
+							machineAnnotationKey: fmt.Sprintf("%s/%s", namespace, "machine"),
+						},
+						Labels: map[string]string{},
+						UID:    "uid",
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind: "Node",
+					},
+				},
+				MHC: mapiv1beta1.MachineHealthCheck{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test",
+						Namespace: namespace,
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind: "MachineHealthCheck",
+					},
+					Spec: mapiv1beta1.MachineHealthCheckSpec{
+						Selector: metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"foo": "bar",
+							},
+						},
+						MaxAge: &metav1.Duration{Duration: time.Second},
```

Comment:
If we can make the machine a day old, we should make this like 20 hours or something, so it's not just any machine that was created. Being a large time difference would make it more obvious what the test is doing rather than a time period that could elapse during the duration of the test.
```suggestion
						MaxAge: &metav1.Duration{Duration: time.Second},
```

Should also add a test case where it hasn't reached the maximum age

Reply:
I added a new test, thanks